GVFS/GVFS.Common/NuGetUpgrader/NuGetUpgrader.cs

```diff
@@ -0,0 +1,531 @@
+                        platformInstallManifest == null)
+                    {
+                        activity.RelatedError($"Extracted ReleaseManifest from JSON, but there was no entry for {platformKey}.");
+                        error = $"No entry in the manifest for the current platform ({platformKey}). Please verify the upgrade package.";
+                        return false;
+                    }
+
+                    activity.RelatedInfo($"Extracted ReleaseManifest from JSON. InstallActions: {platformInstallManifest.InstallActions.Count}");
+
+                    foreach (InstallActionInfo entry in platformInstallManifest.InstallActions)
+                    {
+                        string installerPath = Path.Combine(extractedPackagePath, ContentDirectoryName, entry.InstallerRelativePath);
+
+                        string args = entry.Args ?? string.Empty;
+
+                        // Replace tokens on args
+                        string processedArgs = NuGetUpgrader.ReplaceArgTokens(args, this.InstallationId);
+
+                        activity.RelatedInfo(
+                            "Running install action: Name: {0}, Version: {1}, InstallerPath: {2} RawArgs: {3}, ProcessedArgs: {4}",
+                            entry.Name,
+                            entry.Version,
+                            installerPath,
+                            args,
+                            processedArgs);
+
+                        string progressMessage = string.IsNullOrWhiteSpace(entry.Version) ?
+                            $"Running {entry.Name}" :
+                            $"Running {entry.Name} (version {entry.Version})";
+
+                        installActionWrapper(
+                            () =>
+                            {
+                                if (!this.dryRun)
+                                {
+                                    this.RunInstaller(installerPath, processedArgs, out installerExitCode, out localError);
+                                }
+                                else
+                                {
+                                    // We add a sleep here to ensure
+                                    // the message for this install
+                                    // action is written to the
+                                    // console.  Even though the
+                                    // message is written with a delay
+                                    // of 0, the messages are not
+                                    // always written out.  If / when
+                                    // we can ensure that the message
+                                    // is written out to console, then
+                                    // we can remove this sleep.
+                                    Thread.Sleep(1500);
```

Comment:
File a work item for this?  Assuming this is an issue with the console spinner I think all we'd need to do is add a reset even that ensures the `action()` isn't run until the initial message has been printed.